service-workers/service-worker/active.https.html

```diff
@@ -5,51 +5,55 @@
 <script src="resources/test-helpers.sub.js"></script>
 <body>
 <script>
+
+const SCRIPT = 'resources/empty-worker.js';
+const SCOPE = 'resources/blank.html';
+let registration;
+
 // "active" is set
-async_test(function(t) {
-    var step = t.step_func.bind(t);
-    var url = 'resources/empty-worker.js';
-    var scope = 'resources/blank.html';
-    var frame;
-    var registration;
+promise_test(async t => {
+    let step = t.step_func.bind(t);
+    let frame;
+
+    await service_worker_unregister(t, SCOPE);
+    frame = await with_iframe(SCOPE);
+    registration =
+        await navigator.serviceWorker.register(SCRIPT, {scope: SCOPE});
+    await wait_for_state(t, registration.installing, 'activating');
+    const container = await frame.contentWindow.navigator.serviceWorker;
+    assert_equals(
+        container.controller,
+        null,
+        'On activating state a document should not have a controller');
+    assert_equals(
+        registration.active.scriptURL,
+        normalizeURL(SCRIPT),
+        'On activating state a document should have an active worker ');
+    assert_equals(
+        registration.waiting,
+        null,
+        'On activating state a document should not have a waiting worker');
+    assert_equals(
+        registration.installing,
+        null,
+        'On activating state a document should not have an installing worker');
 
-    service_worker_unregister(t, scope)
-      .then(step(function() { return with_iframe(scope); }))
-      .then(step(function(f) {
-          frame = f;
-          return navigator.serviceWorker.register(url, {scope: scope});
-        }))
-      .then(step(function(r) {
-          registration = r;
-          return wait_for_state(t, r.installing, 'activating');
-        }))
-      .then(step(function() {
-          var container = frame.contentWindow.navigator.serviceWorker;
-          assert_equals(
-            container.controller,
-            null,
-            'On activating state a document should not have a controller');
-          assert_equals(
-            registration.active.scriptURL,
-            normalizeURL(url),
-            'On activating state a document should have an active worker ');
-          assert_equals(
-            registration.waiting,
-            null,
-            'On activating state a document should not have a waiting worker');
-          assert_equals(
-            registration.installing,
-            null,
-            'On activating state a document should not have an installing ' +
-                'worker');
+    // FIXME: Add a test for a frame created after installation.
+    // Should the existing frame ("frame") block activation?
+    t.add_cleanup(() => frame.remove());
+    await registration.unregister();
+}, 'active is set');
```

Comment:
service worker tests tend to use 2 space indents and 4 spaces for wrapping. Could we follow that here?